src/graphnet/models/task/task.py

```diff
@@ -22,14 +25,60 @@ class Task(LightningModule):
     def nb_inputs(self) -> int:
         """Number of inputs assumed by task."""
 
-    def __init__(self, hidden_size: int, target_label: str, loss_function: LossFunction):
+    def __init__(
+        self, 
+        hidden_size: int, 
+        target_label: str, 
+        loss_function: LossFunction,
+        transform_prediction_and_target: Optional[Callable] = None,
+        transform_target: Optional[Callable] = None,
+        transform_inference: Optional[Callable] = None,
+        transform_support: tuple = None,
+    ):
         # Base class constructor
         super().__init__()
 
+        # Check(s)
```

Comment:
Given how complex checking the `transform_*` argument have become, I would perhaps suggest writing just
```python
self._transform_prediction_training = None
self._transform_prediction_inference = None
self._transform_target = None
self._validate_and_set_transforms(
    transform_prediction_and_target,
    transform_target,
    transform_inference,
    transform_support,
)
```
in the constructor and then moving the current functionality to a dedicated utility method.

Reply:
Very good, this has been done and added to the PR!